zcash_primitives/CHANGELOG.md

```diff
@@ -6,13 +6,39 @@ and this library adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `zcash_primitives::legacy::keys`:
+  - `impl From<TransparentKeyScope> for bip32::ChildNumber`
+  - `impl From<NonHardenedChildIndex> for bip32::ChildNumber`
+  - `impl TryFrom<bip32::ChildNumber> for NonHardenedChildIndex`
+
 ### Changed
 - MSRV is now 1.70.0.
+- Bumped dependencies to `secp256k1 0.27`.
+- `zcash_primitives::legacy::keys`:
+  - `AccountPrivKey::{from_bytes, to_bytes}` now use the byte encoding from the
+    inside of a `xprv` Base58 string encoding from BIP 32, excluding the prefix
+    bytes (i.e. starting with `depth`).
+  - `AccountPrivKey::from_extended_privkey` now takes
+    `bip32::ExtendedPrivateKey<secp256k1::SecretKey>`.
+  - The following methods now return `Result<_, bip32::Error>`:
+    - `AccountPrivKey::from_seed`
+    - `AccountPrivKey::derive_secret_key`
+    - `AccountPrivKey::derive_external_secret_key`
+    - `AccountPrivKey::derive_internal_secret_key`
+    - `AccountPubKey::derive_external_ivk`
+    - `AccountPubKey::derive_internal_ivk`
```

Comment:
Would this be a good time to move the transparent keys to `zcash_keys`? I think that if we did so, `zcash_keys` would no longer need to depend upon `zcash_primitives`.

Reply:
Hmm, I kinda see it, but also the other protocol-specific key types exist in their own crates, so `zcash-keys` may not be where we want this code.

Reply:
It may instead make more sense to plan a move to some `zcash_transparent` crate that involves not just the key material, but the transparent builder logic etc. as well.

Reply:
In some sense bip32 *is* that protocol-specific key derivation crate; we just have to add some more layers for Zcash, so I think this going in zcash_keys would be reasonable.

Reply:
Moving these as-is may mean that `zcash_keys` no longer depends on `zcash_primitives`, but it would mean that `zcash_primitives` now needs to depends on `zcash_keys`, because the rest of the transparent logic is still there. So the move would not decouple the crates, it would just invert their relationship.